### PR TITLE
Cody: improve onboarding process

### PR DESCRIPTION
**Pull request by abeatrix**
Close: https://github.com/sourcegraph/sourcegraph/issues/50713#event-9023253287
RE: [issue reported by user](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1681729029259269)
See Loom for detailed walkthrough of the changes included in this PR: https://www.loom.com/share/afd9e7e70bfa454881c61119aea2d8ed

Changes:
- Update UI based on [figma](https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-%5BMain%5D?node-id=2649-12274&t=HdT9rlvaffeyCB0k-0) designed by @danielmarquespt 
- Set `cody.activated` to true only if the user is logged in with valid login credentials 
- Validate login on webview load (previously, we log users in as long as there is an access token)
- Validate login on configs change (RE [slack discussion](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1681731282332559?thread_ts=1681729029.259269&cid=C04MSD3DP5L))
- Add One-click login for Community (sourcegraph.com) users
  image
- Display log-in message in the menu when trying to use right-click menu without logging in:
  image
- Display recipes to logged-in users only
- Show button to `start a new conversation` only when there is an ongoing conversation
- Move settings button to top
- Fix: do not submit chat on send button click when the input is empty

## Test plan

See loom for details: https://www.loom.com/share/afd9e7e70bfa454881c61119aea2d8ed

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | -0.00% (-0.00 kb) | -0.00% (-0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/bee/cody-onboarding/compare-report.html;tab=live_object) for a full comparison between the commits 1f8209c and a53643b or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/cody/webviews/Login.tsx [hunk 1]
```diff
@@ -26,39 +31,70 @@ export const Login: React.FunctionComponent<React.PropsWithChildren<LoginProps>>
...
+                        Click here to request access.
+                    </a>
+                </p>
+                <a href="https://sourcegraph.com/user/settings/tokens/new/callback?requestFrom=CODY">
+                    <VSCodeButton
+                        className={styles.button}
+                        type="button"
+                        onClick={() => setEndpoint('https://sourcegraph.com')}
+                    >
+                        Continue with sourcegraph.com
```
Comment (dominiccooney) [inline client/cody/webviews/Login.tsx:93]: This is definitely an improvement, but I wonder if ordinary people understand what this versus other endpoints means.

File: client/cody/webviews/Login.tsx [hunk 2]
```diff
@@ -26,39 +31,70 @@ export const Login: React.FunctionComponent<React.PropsWithChildren<LoginProps>>
...
+            </form>
+            <p className={styles.inputLabel}>
+                <i className="codicon codicon-account" />
+                <span>Community User</span>
+            </p>
+            <div className={styles.wrapper}>
+                <span>Connect Cody to sourcegraph.com in browser.</span>
+                <p className={styles.input}>
+                    <a href="https://docs.google.com/forms/d/e/1FAIpQLScSI06yGMls-V1FALvFyURi8U9bKRTSKPworBhzZEHDQvo0HQ/viewform">
+                        Click here to request access.
```
Comment (dominiccooney) [inline client/cody/webviews/Login.tsx:84]: Per content guidelines for links, should we be more specific, for example "Fill out this form to request access."
https://handbook.sourcegraph.com/company-info-and-process/communication/content_guidelines/actionable_language/#links

**Review by dominiccooney — APPROVED** [review]
Looks great.

**Review by eseliger — APPROVED** [review]

File: client/cody/webviews/Login.tsx (hunk 1 above)
Comment (abeatrix) [reply client/cody/webviews/Login.tsx:93]: @dominiccooney That's a good point! Will work with @danielmarquespt on the design for the next fine-tuning step to address this. Thanks for the quick review @dominiccooney !!

**Comment by sqs** [comment]
@abeatrix This is super awesome, thanks! I have some small tweaks to the language at https://github.com/sourcegraph/sourcegraph/pull/50773.

**Comment by mrnugget** [comment]
Very cool Loom, @abeatrix!

**Pull request merged**